built model with input shape = (28, 28, 1, 1)
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 conv2d (Conv2D)             (None, 26, 26, 32)        320
 max_pooling2d (MaxPooling2D  (None, 13, 13, 32)       0
 )
 flatten (Flatten)           (None, 5408)              0
 dense (Dense)               (None, 100)               540900
 dense_1 (Dense)             (None, 10)                1010
 softmax (Softmax)           (None, 10)                0
=================================================================
Total params: 542,230
Trainable params: 542,230
Non-trainable params: 0
_________________________________________________________________
There are  188  batches in the warm start
Started Training
2022-08-24 15:59:37.806905: W tensorflow/core/kernels/data/cache_dataset_ops.cc:856] The calling iterator did not fully read the dataset being cached. In order to avoid unexpected truncation of the dataset, the partially cached contents of the dataset  will be discarded. This can happen if you have an input pipeline similar to `dataset.cache().take(k).repeat()`. You should use `dataset.take(k).cache().repeat()` instead.
Traceback (most recent call last):
  File "Main.py", line 83, in <module>
    sf.k_diversity(model,train_ds,config['k'],config['batch_size'],config['alpha'],config['beta'],conn)
  File "/com.docker.devenvironments.code/supporting_functions.py", line 270, in k_diversity
    aprox_grads.append(grads_to_add.flatten) # list of the grads [[grads],[grads],...]
AttributeError: 'list' object has no attribute 'flatten'
(1, 28, 28, 1)
tf.Tensor([4], shape=(1,), dtype=int64)
1/1 [==============================] - 0s 89ms/step
[[-0.0132006   0.02811801 -0.00229798 -0.00367997 -0.01213227 -0.01726447
   0.01381963  0.00771591  0.00601847  0.00325467]]